packages/@aws-cdk-testing/framework-integ/test/custom-resources/test/custom-resource-config/integ.custom-resource-config-logRetention.ts

```diff
@@ -0,0 +1,36 @@
+import * as s3 from 'aws-cdk-lib/aws-s3';
+import * as logs from 'aws-cdk-lib/aws-logs';
+import * as cdk from 'aws-cdk-lib';
+import * as integ from '@aws-cdk/integ-tests-alpha';
+import { Construct } from 'constructs';
+import * as s3deploy from 'aws-cdk-lib/aws-s3-deployment';
+import { CustomResourceConfig } from 'aws-cdk-lib/custom-resources';
+
+class TestCustomResourceConfig extends cdk.Stack {
+  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+
+    const locally_set_log_retention = logs.RetentionDays.ONE_WEEK;
+
+    let websiteBucket = new s3.Bucket(this, 'WebsiteBucket', {});
+
+    new s3deploy.BucketDeployment(this, 's3deployNone', {
+      sources: [s3deploy.Source.jsonData('file.json', { a: 'b' })],
+      destinationBucket: websiteBucket,
+      logRetention: locally_set_log_retention,
+    });
+
+  }
+}
+
+const app = new cdk.App();
+const set_log_retention = logs.RetentionDays.TEN_YEARS;
+CustomResourceConfig.of(app).addLogRetentionLifetime(set_log_retention);
+const testCase = new TestCustomResourceConfig(app, 'test-custom-resource-config-logRetention');
+
+new integ.IntegTest(app, 'integ-test-custom-resource-config-logRetention', {
+  testCases: [testCase],
+  diffAssets: false,
+});
+
+app.synth();
```

Comment:
you don't need the `app.synth()` calls here. This is a leftover requirement from old-style integration tests, which you do not need when you use `IntegTest`.